backend/asmlink.ml

```diff
@@ -439,16 +446,19 @@ let link ~ppf_dump objfiles output_name =
     Clflags.ccobjs := !Clflags.ccobjs @ !lib_ccobjs;
     Clflags.all_ccopts := !lib_ccopts @ !Clflags.all_ccopts;
                                                  (* put user's opts first *)
+    let named_startup_file =
+      !Clflags.keep_startup_file || !Emitaux.binary_backend_available
+    in
     let startup =
-      if !Clflags.keep_startup_file || !Emitaux.binary_backend_available
+      if named_startup_file
```

Comment:
(Maybe do the same on line 380,
or define a helper function.)

Reply:
Fixed